jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannel.java

```diff
@@ -670,6 +670,26 @@ private void dispatch(Dispatchable dispatchable) throws Exception
         }
     }
 
+    private void errorDispatch(Dispatchable dispatchable) throws Exception
```

Comment:
I'm not a fan of these two dispatch methods, that are substantially the same.

Whilst this is used for error dispatching, the name really doesn't say what it does.
This really should be called `dispatch` and the other `dispatchWithServletListeners`

Or maybe just have one `dispatch` method that takes a boolean saying if servlet listeners should be called or not.

Or maybe one method calls the other?


Best option would be to arrange for the listeners to be called elsewhere as we discussed... that didn't work??

Reply:
Also... remind me how servlet listeners are meant to be called for async requests?  If startAsync is called, should we be calling destroy in the finally block?

Reply:
> Best option would be to arrange for the listeners to be called elsewhere as we discussed... that didn't work??

I left a comment above describing why this cannot be done as we discussed before, it really needs to be done in the ServletChannel because during include and forward dispatches we don't want to be calling the listener multiple times.

I will add the boolean for this method.

> Also... remind me how servlet listeners are meant to be called for async requests? If startAsync is called, should we be calling destroy in the finally block?

No not when startAsync is called, only when it exits the filterChain.
```
A ServletRequest is defined as coming into scope of a web application when it is about 
to enter the first servlet or filter of the web application, and as going out of scope as it 
exits the last servlet or the first filter in the chain.
```